app_dart/lib/src/service/firestore.dart

```diff
@@ -111,38 +112,42 @@ class FirestoreService {
 }
 
 /// Generates task documents based on targets.
-List<Document> targetsToTaskDocuments(Commit commit, List<Target> targets) {
-  final Iterable<Document> iterableDocuments = targets.map(
-    (Target target) => Document(
-      name: '$kDatabase/documents/$kTaskCollectionId/${commit.sha}_${target.value.name}_1',
-      fields: <String, Value>{
-        kTaskCreateTimestampField: Value(integerValue: commit.timestamp!.toString()),
-        kTaskEndTimestampField: Value(integerValue: kTaskDefaultTimestampValue.toString()),
-        kTaskBringupField: Value(booleanValue: target.value.bringup),
-        kTaskNameField: Value(stringValue: target.value.name),
-        kTaskStartTimestampField: Value(integerValue: kTaskDefaultTimestampValue.toString()),
-        kTaskStatusField: Value(stringValue: Task.statusNew),
-        kTaskTestFlakyField: Value(booleanValue: false),
-        kTaskCommitShaField: Value(stringValue: commit.sha),
-      },
+List<firestore.Task> targetsToTaskDocuments(Commit commit, List<Target> targets) {
+  final Iterable<firestore.Task> iterableDocuments = targets.map(
+    (Target target) => firestore.Task.fromDocument(
+      taskDocument: Document(
+        name: '$kDatabase/documents/$kTaskCollectionId/${commit.sha}_${target.value.name}_1',
```

Comment:
From looking at the other pull requests related to this effort the '_1' signifies the run attempt correct? And if that is accurate could we maybe add a constant for the initial value like ATTEMPT_INITIAL_VALUE?

Reply:
Good point. Added.